samples: matter: Add guide on how to flash factory data on nRF54H (#18398)
nrfconnect/sdk-nrf: samples/matter/lock/README.rst

.. matter_door_lock_sample_factory_data_end

.. matter_door_lock_sample_factory_data_nrf54h20_start

> Merging the factory data hex file with the firmware hex file is currently not available on nRF54H20 DK.

Reviewer: ```suggestion
Merging the factory data hex file with the firmware hex file is currently not available on the nRF54H20 DK.
```

> The factory data support is disabled by default on this board, however you can still use it, but you need to flash the :file:`factory_data.hex` file manually.

Reviewer: ```suggestion
The factory data support is disabled by default on this board.
You can still use it, but you need to flash the :file:`factory_data.hex` file manually.
```

> To do it complete the following steps in the given order:

Reviewer: ```suggestion
To use factory data on the nRF54H20 DK, complete the following steps:
```

1. Flash the :file:`factory_data.hex` file into the device using the following command:

   .. code-block:: console

      nrfutil device program --firmware build/template/zephyr/factory_data.hex

#. Enable factory data support by building the sample with the :kconfig:option:`CONFIG_CHIP_FACTORY_DATA` Kconfig option set to ``y`` using the following command:

   .. code-block:: console

      west build -p -b nrf54h20dk/nrf54h20/cpuapp -- -DCONFIG_CHIP_FACTORY_DATA=y

> #. Flash the firmware:

Reviewer: ```suggestion
#. Flash the firmware into the device:
```

   .. code-block:: console

      west flash --erase

The ``west flash --erase`` command does not clear the factory data partition.

> If you want to clear it, use the following command, and fill the ``<address>`` argument as an address of the factory data partition, by default it is set to ``0xe174000``:

Reviewer: ```suggestion
If you want to clear it, use the following command, and fill the ``<address>`` argument as an address of the factory data partition.
By default, the address is set to ``0xe174000``.
```
Could also be that the argument defaults to the value, not sure.

Author: They can modify that value in the dts file, so maybe we should do it generically but point out the default value to ensure that it will work for the first attempt.
